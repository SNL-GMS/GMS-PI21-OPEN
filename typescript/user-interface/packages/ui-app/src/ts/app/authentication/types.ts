import type { AuthenticationStatus } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<AuthenticationStatus>;
  checkIsAuthenticated(): Promise<AuthenticationStatus>;
  unAuthenticateWith(): Promise<AuthenticationStatus>;
  logout(setAppAuthenticationStatus: (status: AuthenticationStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
