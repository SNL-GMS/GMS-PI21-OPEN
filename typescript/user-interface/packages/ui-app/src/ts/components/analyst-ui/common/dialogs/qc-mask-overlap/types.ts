import type { QcMaskTypes } from '@gms/common-model';

import type { CoordinatesPx } from '../qc-mask-form/types';
import type { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
