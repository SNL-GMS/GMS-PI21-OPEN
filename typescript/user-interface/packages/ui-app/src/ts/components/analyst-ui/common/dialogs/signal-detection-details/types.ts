import type { SignalDetectionTypes } from '@gms/common-model';
import type { Row } from '@gms/ui-core-components';

/**
 * SignalDetectionDetails Props
 */
export interface SignalDetectionDetailsProps {
  detection: SignalDetectionTypes.SignalDetection;
  color: string;
  assocStatus?: string;
}

/**
 * SignalDetectionDetails State
 */
export interface SignalDetectionDetailsState {
  showHistory: boolean;
}

/**
 * Interface that describes the Detection history
 * information.
 */
export interface SignalDetectionHistoryRow extends Row {
  id: string;
  versionId: string;
  creationTimestamp: number;
  phase: string;
  arrivalTimeMeasurementFeatureType: string;
  arrivalTimeMeasurementTimestamp: number;
  arrivalTimeMeasurementUncertaintySec: number;
  channelSegmentType: string;
  authorName: string;
  rejected: string;
}
