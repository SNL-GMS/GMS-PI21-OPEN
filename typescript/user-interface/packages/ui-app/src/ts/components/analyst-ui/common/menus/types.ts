export interface HideStationMenuItemProps {
  stationName: string;
  hideStationCallback: (stationName: any) => void;
  showHideText?: string;
  disabled?: boolean;
}
