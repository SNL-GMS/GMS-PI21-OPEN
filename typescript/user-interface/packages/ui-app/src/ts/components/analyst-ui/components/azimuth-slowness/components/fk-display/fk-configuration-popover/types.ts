import type { FkTypes } from '@gms/common-model';
import type { CheckboxSearchListTypes } from '@gms/ui-core-components';

import type {
  FkConfigurationWithUnits,
  FkUnits
} from '~analyst-ui/components/azimuth-slowness/types';

export enum FkConfigurationPopoverPanel {
  DEFAULT,
  ADVANCED
}

export interface FkConfigurationPopoverState {
  openPanel: FkConfigurationPopoverPanel;
  fkUnits: FkUnits;
  normalizeWaveforms: boolean;
  mediumVelocity: number;
  maximumSlowness: number;
  numberOfPoints: number;
  channelCheckboxes: CheckboxSearchListTypes.CheckboxItem[];
  useVerticalChannelOffsets: boolean;
}
export interface FkConfigurationPopoverProps extends FkTypes.FkConfiguration {
  fkUnitDisplayed: FkUnits;
  applyFkConfiguration(configuration: FkConfigurationWithUnits);
  close();
}
