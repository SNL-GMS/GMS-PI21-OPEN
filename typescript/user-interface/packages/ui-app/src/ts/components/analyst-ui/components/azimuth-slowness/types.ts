import type {
  CommonTypes,
  FkTypes,
  LegacyEventTypes,
  ProcessingStationTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import type GoldenLayout from '@gms/golden-layout';
import type {
  AnalystWorkspaceTypes,
  ProcessingAnalystConfigurationQueryProps
} from '@gms/ui-state';
import type Immutable from 'immutable';

import type {
  FilterType,
  FkThumbnailSize
} from './components/fk-thumbnail-list/fk-thumbnails-controls';
import type { LeadLagPairs } from './constants';

export interface FkParams {
  windowParams: FkTypes.WindowParameters;
  frequencyPair: FkTypes.FrequencyBand;
}

export interface LeadLagPairAndString {
  leadLagPairs: LeadLagPairs;
  windowParams: FkTypes.WindowParameters;
}

export enum FkUnits {
  FSTAT = 'FSTAT',
  POWER = 'POWER'
}

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export interface FkConfigurationWithUnits extends FkTypes.FkConfiguration {
  fkUnitToDisplay: FkUnits;
}

/**
 * Azimuth Slowness Redux Props
 */
export interface AzimuthSlownessReduxProps {
  // passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  selectedSdIds: string[];
  openEventId: string;
  unassociatedSDColor: string;
  sdIdsToShowFk: string[];
  analysisMode: WorkflowTypes.AnalysisMode;
  location: AnalystWorkspaceTypes.LocationSolutionState;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  channelFilters: Record<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  setSelectedSdIds(ids: string[]): void;
  setChannelFilters(filters: Record<string, WaveformTypes.WaveformFilter>): void;
  setSdIdsToShowFk(signalDetectionIds: string[]): void;
  setMeasurementModeEntries(entries: Record<string, boolean>): void;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * Azimuth Slowness State
 */
export interface AzimuthSlownessState {
  fkThumbnailSizePx: FkThumbnailSize;
  fkThumbnailColumnSizePx: number;
  filterType: FilterType;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  userInputFkFrequency: FkTypes.FrequencyBand;
  numberOfOutstandingComputeFkMutations: number;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  fkInnerContainerWidthPx: number;
  fkFrequencyThumbnails: Immutable.Map<string, FkTypes.FkFrequencyThumbnail[]>;
}

/**
 * Mutations used by the Az Slow display
 */
export interface AzimuthSlownessMutations {
  computeFks: (args: any) => Promise<void>;
  computeFkFrequencyThumbnails: (args: any) => Promise<void>;
  setWindowLead: (args: any) => Promise<void>;
  markFksReviewed: (args: any) => Promise<void>;
}

/**
 * Consolidated props for Azimuth Slowness
 */
export type AzimuthSlownessProps = AzimuthSlownessReduxProps &
  AzimuthSlownessMutations &
  ProcessingAnalystConfigurationQueryProps;

/**
 * State of the az slow panel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
}

export interface AzimuthSlownessPanelProps {
  // Data
  defaultStations: ProcessingStationTypes.ProcessingStation[];
  eventsInTimeRange: LegacyEventTypes.Event[];
  displayedSignalDetection: SignalDetectionTypes.SignalDetection | undefined;
  openEvent: LegacyEventTypes.Event | undefined;
  unassociatedSignalDetectionByColor: string;
  associatedSignalDetections: SignalDetectionTypes.SignalDetection[];
  signalDetectionsToDraw: SignalDetectionTypes.SignalDetection[];
  signalDetectionsIdToFeaturePredictions: Immutable.Map<
    string,
    LegacyEventTypes.FeaturePrediction[]
  >;
  signalDetectionsByStation: SignalDetectionTypes.SignalDetection[];
  featurePredictionsForDisplayedSignalDetection: LegacyEventTypes.FeaturePrediction[];
  distances: LegacyEventTypes.LocationToStationDistance[];
  // Redux state as props
  selectedSdIds: string[];
  defaultWaveformFilters: WaveformTypes.WaveformFilter[];
  sdIdsToShowFk: string[];
  location: AnalystWorkspaceTypes.LocationSolutionState;
  fkFrequencyThumbnails: FkTypes.FkFrequencyThumbnail[];
  // Azimuth display state as props
  fkThumbnailColumnSizePx: number;
  fkDisplayWidthPx: number;
  fkDisplayHeightPx: number;
  filterType: FilterType;
  fkThumbnailSizePx: FkThumbnailSize;
  fkUnitsForEachSdId: Immutable.Map<string, FkUnits>;
  numberOfOutstandingComputeFkMutations: number;
  userInputFkFrequency: FkTypes.FrequencyBand;
  fkUnitForDisplayedSignalDetection: FkUnits;
  userInputFkWindowParameters: FkTypes.WindowParameters;
  fkInnerContainerWidthPx: number;
  channelFilters: Record<string, WaveformTypes.WaveformFilter>;
  selectedSortType: AnalystWorkspaceTypes.WaveformSortType;
  defaultSignalDetectionPhase?: CommonTypes.PhaseType;
  // Prop functions
  adjustFkInnerContainerWidth(
    fkThumbnailsContainer: HTMLDivElement,
    fkThumbnailsInnerContainer: HTMLDivElement
  ): void;
  markFksForSdIdsAsReviewed(sdIds: string[]): void;
  updateFkThumbnailSize(size: FkThumbnailSize): void;
  updateFkFilter(filterType: FilterType): void;
  setFkThumbnailColumnSizePx(newSizePx: number): void;
  computeFkAndUpdateState(fkInput: FkTypes.FkInput): void;
  changeUserInputFks(
    windowParams: FkTypes.WindowParameters,
    frequencyBand: FkTypes.FrequencyBand
  ): void;
  setFkUnitForSdId(sdId: string, fkUnit: FkUnits): void;

  // Redux functions
  setSelectedSdIds(sdIds: string[]): void;
  setSdIdsToShowFk(sdIds: string[]): void;
  setChannelFilters(filters: Record<string, WaveformTypes.WaveformFilter>): void;
  setMeasurementModeEntries(entries: Record<string, boolean>): void;
}
