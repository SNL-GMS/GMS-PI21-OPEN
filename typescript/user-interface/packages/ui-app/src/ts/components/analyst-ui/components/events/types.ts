import type { EventTypes } from '@gms/common-model';
import type { CellRendererParams, Row } from '@gms/ui-core-components';
import { EventsColumn } from '@gms/ui-state';
import Immutable from 'immutable';

/**
 * Event Table generic CellRendererParams type
 */
export type EventTableCellRendererParams<T> = CellRendererParams<
  EventRow,
  unknown,
  T,
  unknown,
  unknown
>;

/**
 * used to match the display strings to values in the site table column picker dropdown
 */
export const columnDisplayStrings: Immutable.Map<EventsColumn, string> = Immutable.Map<
  EventsColumn,
  string
>([
  [EventsColumn.conflict, 'Conflict'],
  [EventsColumn.time, 'Time'],
  [EventsColumn.latitudeDegrees, 'Lat (°)'],
  [EventsColumn.longitudeDegrees, 'Lon (°)'],
  [EventsColumn.depthKm, 'Depth (km)'],
  [EventsColumn.magnitudeMb, 'mb'],
  [EventsColumn.magnitudeMs, 'ms'],
  [EventsColumn.magnitudeMl, 'ml'],
  [EventsColumn.coverageSemiMajorAxis, 'Coverage semi-major'],
  [EventsColumn.coverageSemiMinorAxis, 'Coverage semi-minor'],
  [EventsColumn.confidenceSemiMajorAxis, 'Confidence semi-major'],
  [EventsColumn.confidenceSemiMinorAxis, 'Confidence semi-minor'],
  [EventsColumn.region, 'Region'],
  [EventsColumn.activeAnalysts, 'Active analysts'],
  [EventsColumn.preferred, 'Preferred in stage'],
  [EventsColumn.status, 'Workflow status'],
  [EventsColumn.rejected, 'Rejected']
]);

/**
 * Enumerated time types based on whether event or signal detection is within the interval or a before/after edge event
 */
export enum EdgeTypes {
  BEFORE = 'Before',
  AFTER = 'After',
  INTERVAL = 'Interval'
}

/**
 * Event row for event display
 */
export interface EventRow extends Row {
  readonly id: string;
  readonly edgeEventType: EdgeTypes;
  readonly isOpen: boolean;
  readonly conflict: boolean;
  readonly time: number;
  readonly latitudeDegrees: number;
  readonly longitudeDegrees: number;
  readonly depthKm: number;
  readonly region: string;
  readonly confidenceSemiMajorAxis: number;
  readonly confidenceSemiMinorAxis: number;
  readonly coverageSemiMajorAxis: number;
  readonly coverageSemiMinorAxis: number;
  readonly magnitudeMb: number;
  readonly magnitudeMs: number;
  readonly magnitudeMl: number;
  readonly activeAnalysts: string[];
  readonly preferred: string;
  readonly status: EventTypes.EventStatus;
  readonly rejected: string;
}
