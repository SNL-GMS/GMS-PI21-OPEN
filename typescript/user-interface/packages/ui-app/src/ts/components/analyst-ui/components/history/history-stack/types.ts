import type { CacheTypes } from '@gms/common-model';

/**
 * HistoryStack State
 */
export interface HistoryStackState {
  isEventMode: boolean;
}

/**
 * The types of history entry actions, namely undo and redo.
 */
export enum HistoryEntryAction {
  undo = 'Undo',
  redo = 'Redo'
}

export type HistoryEntryComponentData = HistoryInteractionHandlers & HistoryEntryDisplayFlags;

type IntermediateHistoryEntryProps = HistoryEntryComponentData & CacheTypes.History;

/**
 * The data for a single history list element
 */
export interface HistoryEntryProps extends IntermediateHistoryEntryProps {
  message: string;
}

/**
 * HistoryStack Props
 */
export interface HistoryStackProps {
  historyActions: Map<string, (id: string) => void>;
  isIncluded(summary: CacheTypes.History | CacheTypes.HistoryChange): boolean;
}

export interface HistoryInteractionHandlers {
  handleAction?(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
  handleMouseEnter?(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
  handleMouseOut?(e: React.MouseEvent<HTMLDivElement, MouseEvent>): void;
  handleKeyDown?(e: React.KeyboardEvent<HTMLDivElement>): void;
  handleKeyUp?(e: React.KeyboardEvent<HTMLDivElement>): void;
}

export interface HistoryEntryDisplayFlags {
  isAffected?: boolean;
  isAssociated?: boolean;
  isCompleted?: boolean;
  isInConflict?: boolean;
  isOrphaned?: boolean;
  isEventReset?: boolean;
  isChild?: boolean;
  entryType?: HistoryEntryAction;
}

export interface MultipleHistoryEntryProps extends CacheTypes.History {
  description: string;
  historySummary: CacheTypes.History;
  historyEntryChildren: GenericHistoryEntryProps[];
}

export interface GenericHistoryEntryProps extends HistoryEntryComponentData {
  message: string;
}
