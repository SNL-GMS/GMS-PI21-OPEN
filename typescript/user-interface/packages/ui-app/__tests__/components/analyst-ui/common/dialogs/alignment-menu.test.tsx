import { CommonTypes } from '@gms/common-model';
import { PhaseType } from '@gms/common-model/lib/common/types';
import { sleep } from '@gms/common-util';
import { AlignWaveformsOn } from '@gms/ui-state/lib/app/state/analyst/types';
import * as React from 'react';
import { act } from 'react-test-renderer';

import { AlignmentMenu } from '../../../../../src/ts/components/analyst-ui/common/dialogs';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');
// set up window alert and open so we don't see errors
window.alert = jest.fn();
window.open = jest.fn();

describe('Alignment Menu', () => {
  it('Renders', () => {
    const wrapper = Enzyme.mount(
      <AlignmentMenu
        alignedOn={AlignWaveformsOn.PREDICTED_PHASE}
        phaseAlignedOn={CommonTypes.PhaseType.LR}
        defaultPhaseAlignment={CommonTypes.PhaseType.P}
        sdPhases={[CommonTypes.PhaseType.LR, CommonTypes.PhaseType.Lg]}
        prioritySdPhases={[CommonTypes.PhaseType.P]}
        onSubmit={() => {
          jest.fn();
        }}
      />
    );
    expect(wrapper.render()).toMatchSnapshot();
    expect(wrapper.instance().getState()).toEqual({
      alignedOn: 'Predicted',
      phaseAlignedOn: 'LR',
      predictedOrObservedVal: 'Predicted'
    });
    wrapper.unmount();
  });

  it('un-mounts', async () => {
    const onSubmit = jest.fn();
    const wrapper = Enzyme.mount(
      <AlignmentMenu
        alignedOn={AlignWaveformsOn.PREDICTED_PHASE}
        phaseAlignedOn={CommonTypes.PhaseType.LR}
        defaultPhaseAlignment={CommonTypes.PhaseType.P}
        sdPhases={[CommonTypes.PhaseType.LR, CommonTypes.PhaseType.Lg]}
        prioritySdPhases={[CommonTypes.PhaseType.P]}
        onSubmit={onSubmit}
      />
    );
    wrapper.instance().onPhaseSelection(PhaseType.Lg);
    await act(async () => {
      // eslint-disable-next-line @typescript-eslint/no-magic-numbers
      await sleep(2000);
      wrapper.unmount();
    });

    expect(onSubmit).toHaveBeenCalledWith('Predicted', 'Lg');
  });

  it('handles radio on change events', () => {
    const onSubmit = jest.fn();
    const wrapper = Enzyme.mount(
      <AlignmentMenu
        alignedOn={AlignWaveformsOn.PREDICTED_PHASE}
        phaseAlignedOn={CommonTypes.PhaseType.LR}
        defaultPhaseAlignment={CommonTypes.PhaseType.P}
        sdPhases={[CommonTypes.PhaseType.LR, CommonTypes.PhaseType.Lg]}
        prioritySdPhases={[CommonTypes.PhaseType.P]}
        onSubmit={onSubmit}
      />
    );
    const mockEvent: any = { currentTarget: { value: 'time' } };
    wrapper.find('input[data-cy="time"]').first().props().onChange(mockEvent);
    expect(onSubmit).toHaveBeenCalledWith('Time');

    onSubmit.mockClear();

    mockEvent.currentTarget.value = 'phase';

    wrapper.find('[data-cy="phase"]').first().props().onChange(mockEvent);
    expect(onSubmit).toHaveBeenCalledTimes(0);

    wrapper.find('[data-cy="filterable-option-Lg"]').simulate('click');
    expect(onSubmit).toHaveBeenCalledWith('Predicted', 'Lg');
    expect(wrapper.instance().getState()).toEqual({
      alignedOn: 'Predicted',
      phaseAlignedOn: 'Lg',
      predictedOrObservedVal: 'Predicted'
    });
  });
});
