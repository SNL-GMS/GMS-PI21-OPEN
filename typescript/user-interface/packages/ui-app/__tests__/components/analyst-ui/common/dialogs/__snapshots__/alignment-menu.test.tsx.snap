// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alignment Menu Renders 1`] = `
<div
  class="alignment-dropdown"
>
  <div>
    <label
      class="bp4-control bp4-radio alignment-dropdown__radio"
    >
      <input
        data-cy="time"
        name="Blueprint4.RadioGroup-0"
        type="radio"
        value="time"
      />
      <span
        class="bp4-control-indicator"
      />
      Time
    </label>
    <label
      class="bp4-control bp4-radio alignment-dropdown__radio"
    >
      <input
        checked=""
        data-cy="phase"
        name="Blueprint4.RadioGroup-0"
        type="radio"
        value="phase"
      />
      <span
        class="bp4-control-indicator"
      />
      Phase:
      <div
        class="alignment-dropdown__phase-type-selector"
      >
        <div
          class="dropdown-container"
        >
          <span
            class="dropdown-selector"
            data-cy="-dropdown"
          >
            <div
              class="bp4-html-select"
            >
              <select
                style="min-width: 95px; width: 95px;"
                title="Predicted"
              >
                <option
                  value="Predicted"
                >
                  Predicted
                </option>
                <option
                  value="Observed"
                >
                  Observed
                </option>
              </select>
              <span
                class="bp4-icon bp4-icon-double-caret-vertical"
                icon="double-caret-vertical"
              >
                <svg
                  data-icon="double-caret-vertical"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    Open dropdown
                  </desc>
                  <path
                    d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </div>
          </span>
        </div>
      </div>
    </label>
  </div>
  <div
    class="alignment-dropdown__list-positioner"
  >
    <div
      class="filterable-option-list"
    >
      <input
        class="filterable-option-list__search"
        placeholder="Search input"
        style="width: 169px; margin-left: 0px;"
        tabindex="0"
        type="search"
        value=""
      />
      <div
        class="filterable-option-list__list"
        style="width: 169px; margin-left: 0px;"
      >
        <div
          class="filterable-option-list-item"
          data-cy="filterable-option-P"
          id="P"
          tabindex="0"
        >
          P
        </div>
        <div
          class="filterable-option-list__divider"
        />
        <div
          class="filterable-option-list-item"
          data-cy="filterable-option-Lg"
          id="Lg"
          tabindex="0"
        >
          Lg
        </div>
        <div
          class="filterable-option-list-item__selected filterable-option-list-item"
          data-cy="filterable-option-LR"
          id="LR"
          tabindex="0"
        >
          LR
        </div>
      </div>
    </div>
  </div>
</div>
`;
